src/ingest-pipeline/misc/tools/new_dataset_survey.py

```diff
@@ -115,6 +126,114 @@ def join_notes(df, notes_df):
     return df.drop(columns=['note_x', 'note_y'])
 
 
+def get_google_service():
+    creds = None
+    # The file token.json stores the user's access and refresh tokens, and is
+    # created automatically when the authorization flow completes for the first
+    # time.
+    if os.path.exists('.token.json'):
+        creds = Credentials.from_authorized_user_file('.token.json', SCOPES)
+    # If there are no (valid) credentials available, let the user log in.
+    if not creds or not creds.valid:
+        if creds and creds.expired and creds.refresh_token:
+            creds.refresh(Request())
+        else:
+            flow = InstalledAppFlow.from_client_secrets_file(
+                '.sheetreader_key.json', SCOPES)
+            creds = flow.run_local_server(port=0)
```

Comment:
What actually happens when this path executes?  How does the interaction with the user occur, and how can we get Airflow to deal with it?

Reply:
Goes to the authorization part of Google Apps, it opens a web page when you click authorize (if you are actually allowed to see the doc) and generates de authentication token based on the sheetreaders_key.json created for this particular project.

Reply:
Understood, though I don't see how it's going to open a web page if it's running on hivevm191.

Reply:
You are right, it will give you a link for that but the callback won't work. It will need to be created locally and then moved over the server (the file will be .token.json).